lib/unimatrix/authorization/client_credentials_grant.rb

```diff
@@ -25,14 +25,22 @@ def request_token
           if response.code == '200'
             body = JSON.parse( response.body )
             body = body[ 'token' ] if body[ 'token' ].present?
-
-            body[ 'access_token' ] rescue nil
+            result = body[ 'access_token' ] rescue nil
+            if body.include?( 'error' )
+               puts "ERROR: #{ body }"
+               return body
+            end
+            if options == { 'include_expiry' => true }
```

Comment:
I would write this more like `if options['include_expiry'] == true` so that you could theoretically in the future have other keys in the options hash